File: hub/src/light/effect/resolved_effect.py
class ResolvedEffect:
    def __init__(self, rgbw, amplitude, ledPositions):
       self.rgbw = rgbw
       self.amplitude = amplitude
       self.ledPositions = ledPositions

    @staticmethod
    def createRgbw(rgbw):
        return ResolvedEffect(rgbw, None, None)

    @staticmethod
    def createAmplitude(amplitude):
        return ResolvedEffect(None, amplitude, None)

    @staticmethod
    def createTranslation(ledPositions):
        return ResolvedEffect(None, None, ledPositions)

    def toString(self):
        return str([None if self.rgbw is None else self.rgbw.toString(), self.amplitude, self.ledPositions])

File: hub/drumlite-js/src/effect/ResolvedEffect.ts
import RGB from "../light/RGB"

export default class ResolvedEffect {
    rgb?: RGB;
    amplitude?: number;
    ledPositions?: number[];
    
    constructor(rgb?: RGB, amplitude?: number, ledPositions?: number[]) {
       this.rgb = rgb;
       this.amplitude = amplitude;
       this.ledPositions = ledPositions;
    }

    public static createRgb(rgb: RGB) {
        return new ResolvedEffect(rgb, undefined, undefined);
    }

    public static createAmplitude(amplitude: number) {
        return new ResolvedEffect(undefined, amplitude, undefined);
    }

    public static createTranslation(ledPositions: number[]) {
        return new ResolvedEffect(undefined, undefined, ledPositions);
    }

    public toString() {
        return `${[this.rgb ? this.rgb : undefined, this.amplitude, this.ledPositions]}`;
    }
}